src/passthrough/async_io.rs

```diff
@@ -314,6 +314,17 @@ impl<D: AsyncDrive + Sync, S: BitmapSlice + Send + Sync> AsyncFileSystem<D, S>
             })
             .await?;
 
+        let mut attr_flags: u32 = 0;
+        if let Some(dax_file_size) = self.cfg.dax_file_size {
+            // st.stat.st_size is i64
+            if self.perfile_dax.load(Ordering::Relaxed)
+                && st.stat.st_size >= 0x0
```

Comment:
is it possible that `st_size` has negative value?

Reply:
It seems not possible from stat64(2), but I think it's no harm to be defensive.